### PR TITLE
#159 first admin review page

### LINKED ISSUE #159: [Frontend] - Admin Page to Review Datasets
As an admin, I would like a page to review the new datasets uploaded to the database.

Description:
Admins are going to review the new datasets uploaded to the database to see their validity. To be able to do this, there must be a page that has a list of all the new datasets as well as a way to view each one individually, an option to edit the details (incase of small errors that can be fixed without biased), a way to flag an entry (with comment), and an accept/send flags/reject decision.

In the mockups, there is the menu that has the route to the page as well as (X) where X is the number of datasets in need of review. The second mockup image contains the same format as the dataset upload form. This form should be able to be modified. There is also the option to flag the datasets. **This should be a button with a popup to leave a comment.** The flags in the mockup are no longer necessary!

There is also the option on the bottom to rate it based on stars. 1 is the minimum, 5 is max, and an option to add additional comments. The star rating and comments are only available if nothing is flagged. As is the same with the ACCEPT button.

If something is flagged, an option to add a comment to this flag should appear.

**Mockup**
Menu with Notifications
image   
Admin Page
image

Acceptance Criteria:
- List of all new (un-reviewed) datasets
- Option to view each dataset individually -> Map to DataUpload form model
- Option to flag a dataset
- Add comment to the dataset
- Add comment on the flagged
- Acceptance/Flag/Reject Option

### PR DESCRIPTION
**Pull request by Vatika1**
This branch lets the admin to edit, flag, delete or approve a dataset.

approve dataset

https://user-images.githubusercontent.com/35615793/106346780-7819dd00-6287-11eb-84cd-f5a0143585ae.mp4

delete dataset

https://user-images.githubusercontent.com/35615793/106346762-4f91e300-6287-11eb-9a83-161cc63b900f.mp4

### CODE REVIEW

File: Client/src/Common/Consts/Routes.js
Comment (TiraFire) [inline Client/src/Common/Consts/Routes.js:10]: Later we might want to change this to datasetReview, adminReview sounds like we are reviewing admins XD

File: Client/src/Components/Admin/AdminReviewView.tsx
Comment (TiraFire) [inline Client/src/Components/Admin/AdminReviewView.tsx:29]: Why is the additional comments being passed when a dataset is flagged?

File: Client/src/Components/Admin/AdminReviewView.tsx
Comment (TiraFire) [inline Client/src/Components/Admin/AdminReviewView.tsx:36]: We should make sure to add snackbar messages confirming approval, flagging and removal

File: Client/src/Components/Admin/AdminReviewView.tsx
Comment (TiraFire) [inline Client/src/Components/Admin/AdminReviewView.tsx:66]: Are we able to disable the other buttons while in edit mode? Also, i still think that updating a dataset does not need approval. Admin should be able to edit out a typo and still be able to flag it. We should ask Kiyanda for his feedback on monday for this.

File: Client/src/Components/Admin/AdminReviewView.tsx
Comment (TiraFire) [inline Client/src/Components/Admin/AdminReviewView.tsx:102]: We might want to move this to the top of the dataset instead of the bottom, it is not very obvious that it was there.

File: Client/src/Components/Forms/ComponentUpdate.tsx
Comment (TiraFire) [inline Client/src/Components/Forms/ComponentUpdate.tsx:4]: glad you found this solution to the formik issue 👍

File: Client/src/Remote/RemoteHelper.tsx
Comment (TiraFire) [inline Client/src/Remote/RemoteHelper.tsx]: I think we can remove this

File: Server/src/controllers/DataSetController.ts
Comment (TiraFire) [inline Server/src/controllers/DataSetController.ts]: Remove this as well

File: Server/src/routes/DatasetRouter.ts
Comment (TiraFire) [inline Server/src/routes/DatasetRouter.ts]: this too

File: Server/src/routes/DatasetRouter.ts
Comment (TiraFire) [inline Server/src/routes/DatasetRouter.ts]: remove extra line?

**Review by TiraFire — COMMENTED** [review]
Looks good! There are some improvement in the UI that we can do to make it better for UX, but overall the functionality is there so im glad! There are some small things that you should change based on my comments (remove lines etc), but once that is done, ill be glad to approve

File: Client/src/Components/Admin/AdminReviewView.tsx
Comment (lposo) [reply Client/src/Components/Admin/AdminReviewView.tsx:29]: The route is able to take additional comments as a parameter, so they're sent as well.

File: Client/src/Components/Admin/AdminReviewView.tsx
Comment (lposo) [reply Client/src/Components/Admin/AdminReviewView.tsx:36]: Snackbar has been implemented now.

File: Client/src/Components/Admin/AdminReviewView.tsx
Comment (lposo) [reply Client/src/Components/Admin/AdminReviewView.tsx:102]: That'll probably be dealt with when the Dataset Upload Form is refactored. Since currently we're reusing the button that was used to submit a new dataset.

File: Client/src/Components/Admin/AdminReviewView.tsx
Comment (TiraFire) [reply Client/src/Components/Admin/AdminReviewView.tsx:29]: @Arrow105

File: Client/src/Components/Admin/AdminReviewView.tsx
Comment (TiraFire) [reply Client/src/Components/Admin/AdminReviewView.tsx:36]: Awesome

File: Client/src/Components/Admin/AdminReviewView.tsx
Comment (TiraFire) [reply Client/src/Components/Admin/AdminReviewView.tsx:102]: Makes sense

**Review by TiraFire — APPROVED** [review]

File: Client/src/Components/Admin/AdminReviewList.tsx
Comment (ds161833) [inline Client/src/Components/Admin/AdminReviewList.tsx:11]: very confusing variable name

Comment (ds161833) [reply Client/src/Components/Admin/AdminReviewList.tsx:11]: what does it update?

File: Client/src/Components/Admin/AdminReviewList.tsx
Comment (ds161833) [inline Client/src/Components/Admin/AdminReviewList.tsx:26]: rename to dataset e.g.
datasets.map(dataset =>

File: Client/src/Components/Admin/AdminReviewRow.tsx
Comment (ds161833) [inline Client/src/Components/Admin/AdminReviewRow.tsx:26]: could have just wrote onClick={() => onChange(dataset)}

File: Client/src/Components/Admin/AdminReviewView.tsx
Comment (ds161833) [inline Client/src/Components/Admin/AdminReviewView.tsx:11]: if this is an array, call it datasetStates

**Review by ds161833 — APPROVED** [review]

File: Client/src/Components/Admin/AdminReviewView.tsx
Comment (silent-po2) [reply Client/src/Components/Admin/AdminReviewView.tsx:29]: This is part of the stakeholder requirements. He wants the ability to pass those, period.

**Pull request merged**
